Added new Gravity setting

mash8 opened the pull request:
Adding new Global Settings class
Currently has one setting "Gravity"
Changing setting gravity will effect players gravity

mash8 commented on Assembly-CSharp.csproj:
```diff
@@ -494,66 +496,66 @@
     <Compile Include="Assets\Scripts\Events\Args\HeroHitEvent.cs" />
   </ItemGroup>
   <ItemGroup>
-     <None Include="Assets\Resources\shaders\Unlit - Depth Cutout.shader" />
```
Honestly don't know what i did to make this happen,

mash8 commented on Assets/Scripts/Characters/Human/Hero.cs:
```diff
@@ -84,7 +84,8 @@ public class Hero : Human
     private Transform forearmR;
-    private float gravity = 20f;
+    private float Gravity => 20f * gravityModifier;
+    private float gravityModifier = GameSettings.Global.Gravity ?? 1;
```
Ticket said that the Gravity setting had to be nullable, would it be an idea to just give it a default value of 1 so we don't end up having to do null checks like this every time we want to read it?

play-k replied:
wait u can make a custom map and have gravity like the moon? that's poggers

Lithrun commented:
#33 has been merged into `origin/development`. @mash8 please resolve the merge conflicts, and then this one should be ready to get merged as well

Lithrun reviewed (changes requested):
See https://github.com/AoTTG-2/AoTTG-2/pull/551#issuecomment-798911087

mash8 committed:
Adding new Global Settings class
Currently has one setting "Gravity"
Changing setting gravity will effect players gravity

mash8 commented:
> #33 has been merged into `origin/development`. @mash8 please resolve the merge conflicts, and then this one should be ready to get merged as well

Should be good to go now.

Lithrun committed:
Merge /w development

Lithrun committed:
Resolve minor issues
